doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst

```diff
@@ -201,12 +201,29 @@ Automated metadata generation
 By enabling the ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA`` option in sysbuild, the metadata will be automatically parsed from the ``.elf`` and ``.config`` files.
 The parsed data is stored in the :file:`ble_mesh_metadata.json` file.
 The file is placed in the :file:`dfu_application.zip` archive in the build folder of the application.
-Additionally, the metadata string required by the ``mesh models dfu slot add`` command will be printed in the command line window when the application is built::
+Additionally, the metadata string and firmware ID required by the ``mesh models dfu slot add`` command are printed in the command-line window when the application is built::
 
   Bluetooth Mesh Composition metadata generated:
     Encoded metadata: 020000000000000094cf24017c26f3710100
+    Firmware ID: 59000200000000000000
     Full metadata written to: APPLICATION_FOLDER\build\zephyr\dfu_application.zip
 
+You can generate the Firmware ID using one of the following options:
+
+* To use a user-supplied hex-string as the Firmware ID, enable the ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM`` option.
+  This is the default behavior.
+
+  * If this option is selected, the ``SB_CONFIG_DFU_ZIP_BLUETOOTH_MESH_METADATA_FWID_CUSTOM_HEX`` option must be set to a valid Firmware ID.
+    At minimum, it should include a Company ID in little-endian order.
+    The rest of the string is vendor-specific version information.
+  * When using the :ref:`zephyr:bluetooth_mesh_dfd_srv` for distribution, the number of bytes in the Firmware ID of images used during distribution must not exceed the value of the :kconfig:option:`CONFIG_BT_MESH_DFU_FWID_MAXLEN` Kconfig option set on the distributor node.
```

Comment:
Potential task. I think we should not have `CONFIG_BT_MESH_DFU_FWID_MAXLEN` option to begin with, and ideally distributor should support max length as defined by the spec. Because distributor could be used in a mixed ecosystem, and then target won't be upgradable using a specific distributor and a specific vendor app, IF target uses longer FWID. That would be really annoying situation to have to just save 100 or so bytes of RAM.

Reply:
Feel free to create a JIRA task